File: client/services/history.ts
import { api } from "@/config/api";

export const getHistories = async (token: string) => {
  const response = await api.get(`/history/all`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
  return response;
};

export const updateVisibility = async (
  id: string,
  token: string,
  visibility: "public" | "private",
) => {
  const response = await api.put(
    `/history/${id}/visibility`,
    { visibility },
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    },
  );
  return response;
};

export const updateTitle = async (id: string, title: string, token: string) => {
  const response = await api.put(
    `/history/${id}/title`,
    { title },
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    },
  );
  return response;
};

export const deleteHistory = async (id: string, token: string) => {
  const response = await api.delete(`/history/${id}/delete`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
  return response;
};

export const getQueriesByHistory = async (
  token: string,
  historyUrl: string,
) => {
  const response = await api.get(`/history/${historyUrl}`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
  return response;
};

File: server/routes/history.py
from flask import Blueprint
from config.db import get_db
from controllers.history import (
    get_user_histories,
    get_history_by_url,
    update_visibility,
    update_title,
    delete_history,
)

history = Blueprint("history", __name__)
db = get_db()

history.get("/all")(get_user_histories)
history.get("/<string:url>")(get_history_by_url)
history.put("/<id>/visibility")(update_visibility)
history.put("/<id>/title")(update_title)
history.delete("/<id>/delete")(delete_history)
